pybossa/util.py

```diff
@@ -1253,6 +1253,12 @@ def process_table_component(tp_code, user_response):
     for table_element in table_elements:
         response_key = table_element.get('name')
         response_value = user_response.get(response_key, [])
+
+        # In case the response_value is dict, we need to convert it to a list
+        # so that the table-element can display the data correctly
+        if type(response_value) is dict:
+            response_value = list(response_value.values())
```

Comment:
Is it guaranteed that response values are in the order in which table headers are ordered or the code need to build the list in the order the table headers are placed ensuring each data is listed in their respective header columns?

Reply:
Great question! The order can be guaranteed as "Changed in version 3.7: Dictionary order is guaranteed to be insertion order." (source [here](https://docs.python.org/3/library/stdtypes.html#dictionary-view-objects))

Reply:
Thanks for confirming.